packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +290,102 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def dns_managed?
+      @scenario_answers['foreman_proxy']['dns'] &&
+        @scenario_answers['foreman_proxy']['dns_managed']
+    end
+
+    def should_update_dns?
+      dns_managed? && !@options[:skip_dns]
+    end
+
+    def query_dns_server(fqdn, zone, nameserver_ip)
+      query_dns = Resolv::DNS.new(nameserver: [nameserver_ip], search: [zone], ndots: 1)
```

Comment:
IMHO you shouldn't provide a `search` here. Searching is providing `www` and expanding that to `www.$zone` and since the FQDN should be fully qualified, that should never happen. Worst case it can actually introduce bugs.

Reply:
Thanks, updating